src/System.Net.Http/src/System/Net/Http/HPack/StaticTable.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Net.Http.HPack
+{
+    internal class StaticTable
+    {
+        private static readonly StaticTable _instance = new StaticTable();
+
+        private readonly Dictionary<int, int> _statusIndex = new Dictionary<int, int>
+        {
+            [200] = 8,
+            [204] = 9,
+            [206] = 10,
+            [304] = 11,
+            [400] = 12,
+            [404] = 13,
+            [500] = 14,
+        };
+
+        private StaticTable()
+        {
+        }
+
+        public static StaticTable Instance => _instance;
+
+        public int Count => _staticTable.Length;
+
+        public HeaderField this[int index] => _staticTable[index];
+
+        public IReadOnlyDictionary<int, int> StatusIndex => _statusIndex;
```

Comment:
Unless the JIT is able to devirtualize this, exposing the `Dictionary<int, int>` as `IReadOnlyDictionary<int, int>` will cause every access to be an interface dispatch.